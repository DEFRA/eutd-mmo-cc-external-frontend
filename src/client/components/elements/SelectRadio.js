import styled from 'react-emotion';
import Radio from '../elements/radio';
import { YELLOW } from 'govuk-colours';

const SelectRadio = styled(Radio)({
  '& input:checked ~ span::after': {
    opacity: 1
  },
  '& input:focus ~ span::before': {
    boxShadow: `0 0 0 4px ${YELLOW}`
  },
  '& input:disabled ~ span': {
    opacity: '.4',
    pointerEvents: 'none'
  }
});

export default SelectRadio;