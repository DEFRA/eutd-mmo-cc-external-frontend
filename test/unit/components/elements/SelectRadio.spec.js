import {  mount } from 'enzyme';
import * as React from 'react';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion';
import { render } from '@testing-library/react';

import SelectRadio from '../../../../src/client/components/elements/SelectRadio';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('SelectRadio', () => {
  it('should load successfully', () => {
    const comp = mount(
      <SelectRadio name={'sample'} value={'1'}>Boo</SelectRadio>
    );
    expect(comp).toBeDefined();
  });
});

describe('When loading radio on server side rendering', () => {
  it('should override + selectors to select sibling with ~ css selector', () => {
    const { container } = render(<SelectRadio name={'sample'} value={'1'} checked readonly>Boo</SelectRadio>);
    expect(container).toMatchSnapshot();
  });
});