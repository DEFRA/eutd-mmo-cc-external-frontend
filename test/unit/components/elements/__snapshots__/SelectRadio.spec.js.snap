// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading radio on server side rendering should override + selectors to select sibling with ~ css selector 1`] = `
.emotion-2 {
  display: block;
  position: relative;
  padding: 0 0 0 38px;
  margin-right: 0px;
  margin-bottom: 10px;
}

@media only screen and (min-width:641px) {
  .emotion-2 {
    margin-bottom: 10px;
  }
}

.emotion-2 input:checked ~ span::after {
  opacity: 1;
}

.emotion-2 input:focus ~ span::before {
  box-shadow: 0 0 0 4px #ffbf47;
}

.emotion-2 input:disabled ~ span {
  opacity: .4;
  pointer-events: none;
}

.emotion-0 {
  position: absolute;
  cursor: pointer;
  left: 0;
  top: 0;
  width: 38px;
  height: 38px;
  z-index: 1;
  margin: 0;
  zoom: 1;
  opacity: 0;
  cursor: pointer;
}

.emotion-0:checked + span::after {
  opacity: 1;
}

.emotion-0:focus + span::before {
  box-shadow: 0 0 0 4px #ffbf47;
}

.emotion-0 + span {
  opacity: 1;
  pointer-events: auto;
}

.emotion-1 {
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 19px;
  line-height: 1.25;
  cursor: pointer;
  padding: 8px 10px 9px 12px;
  display: block;
}

.emotion-1:before {
  content: '';
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  width: 33px;
  height: 33px;
  border: 2px solid black;
  border-radius: 50%;
  background: transparent;
}

.emotion-1:after {
  content: '';
  position: absolute;
  top: 8px;
  left: 8px;
  width: 0;
  height: 0;
  border: 8.5px solid;
  border-radius: 50%;
  opacity: 0;
}

<div>
  <label
    class="emotion-2"
    id="label-undefined"
  >
    <input
      checked=""
      class="emotion-0"
      name="sample"
      type="radio"
      value="1"
    />
    <span
      class="emotion-1"
    >
      Boo
    </span>
  </label>
</div>
`;
